kong/db/strategies/postgres/connector.lua

```diff
@@ -371,9 +372,46 @@ function _mt:setkeepalive()
 end
 
 
+function _mt:acquire_lock()
```

Comment:
I suggest that we rename this function to something else, such as `acquire_query_resource` or something similar. Otherwise, using `lock` makes it sound like we are employing a some sort of mutex to execute queries.